src/util/file_log.rs

```diff
@@ -57,47 +57,92 @@ fn open_log_file(path: &str) -> io::Result<File> {
+    rotate_size: Option<usize>,
+    rotate_time: Option<Tm>,
     file_path: String,
     file: File,
 }
 
 impl RotatingFileLoggerCore {
-    fn new(path: &str) -> io::Result<RotatingFileLoggerCore> {
+    fn new(
+        path: &str,
+        rotate_type: RotateType,
+        rotate_size: Option<usize>,
+    ) -> io::Result<RotatingFileLoggerCore> {
         let file = open_log_file(path)?;
-        let file_attr = fs::metadata(path).unwrap();
-        let file_modified_time = file_attr.modified().unwrap();
-        let rollover_time = compute_rollover_time(systemtime_to_tm(file_modified_time));
-        let ret = RotatingFileLoggerCore {
-            rollover_time: rollover_time,
+
+        let mut ret = RotatingFileLoggerCore {
+            rotate_type: rotate_type,
+            rotate_size: None,
+            rotate_time: None,
             file_path: path.to_string(),
             file: file,
         };
-        Ok(ret)
+        match rotate_type {
+            RotateType::ByDay => {
+                assert!(rotate_size.is_none());
+                let file_attr = fs::metadata(path).unwrap();
+                let file_modified_time = file_attr.modified().unwrap();
+                let rotate_time = compute_rotate_time(systemtime_to_tm(file_modified_time));
+                ret.rotate_time = Some(rotate_time);
+                Ok(ret)
+            }
+            RotateType::BySize => {
+                assert!(rotate_size.is_some());
+                ret.rotate_size = rotate_size;
+                Ok(ret)
+            }
+        }
     }
 
     fn open(&mut self) {
         self.file = open_log_file(&self.file_path).unwrap()
     }
 
-    fn should_rollover(&mut self) -> bool {
-        time::now() > self.rollover_time
+    fn should_rotate(&mut self) -> bool {
+        match self.rotate_type {
+            RotateType::ByDay => time::now() > *self.rotate_time.as_ref().unwrap(),
+            RotateType::BySize => match fs::metadata(&self.file_path) {
+                Ok(metadata) => metadata.len() as usize >= *self.rotate_size.as_ref().unwrap(),
+                _ => false,
+            },
+        }
     }
 
-    fn do_rollover(&mut self) {
+    fn do_rotate(&mut self, now: Option<Tm> /* this parameter used in test*/) {
         self.close();
         let mut s = self.file_path.clone();
         s.push_str(".");
-        s.push_str(&time::strftime("%Y%m%d", &one_day_before(self.rollover_time)).unwrap());
+        match self.rotate_type {
+            RotateType::ByDay => {
+                s.push_str(
+                    &time::strftime("%Y%m%d", &one_day_before(self.rotate_time.unwrap())).unwrap(),
+                );
+                self.update_rotate_time();
+            }
+            RotateType::BySize => {
+                let now = match now {
+                    Some(now) => now,
+                    None => time::now(),
+                };
+                s.push_str(&time::strftime("%Y%m%d%H%M%S", &now).unwrap());
```

Comment:
can we use a separator to split the date and time, it is difficult to check the file name now. 

IMO, the format "%Y%m%d" or  "%Y%m%d%H%M%S" is compact and not intuitional.

Reply:
I think we should agree on a format for tidb/tikv/pd.

Reply:
Look at `tidb-2018-02-01T10-56-28.376.log`, so ugly - -

Reply:
For simplicity, I think 20180201_105628.log is fine.